light_curves/code_src/ztf_functions.py

```diff
@@ -10,316 +11,91 @@
-    # if your science depends on precise times, this will need to be corrected.
-    ztf_df = ztf_df.rename(columns={"hmjd": "time"})
-
-    # "explode" the data structure into one row per light curve point and set the correct dtypes
-    # note that this operation may require 6+ times the RAM used for ztf_df up to now
-    ztf_df = ztf_df.explode(["time", "mag", "magerr", "catflags"], ignore_index=True)
-    ztf_df = ztf_df.astype({"time": "float", "mag": "float", "magerr": "float", "catflags": "int"})
-
-    # remove data flagged as bad
-    ztf_df = ztf_df.loc[ztf_df["catflags"] < 32768, :]
-
-    # calc flux [https://arxiv.org/pdf/1902.01872.pdf zeropoint corrections already applied]
-    mag = ztf_df["mag"].to_numpy()
-    magerr = ztf_df["magerr"].to_numpy()
-    fluxupper = ((mag - magerr) * u.ABmag).to_value('mJy')
-    fluxlower = ((mag + magerr) * u.ABmag).to_value('mJy')
-    ztf_df["flux"] = (mag * u.ABmag).to_value('mJy')
-    ztf_df["err"] = (fluxupper - fluxlower) / 2
+    # 1) fetch the raw arrays 
+    raw = hats_get_lightcurves(
+        sample_table,
+        object_catalog_path=UPath('s3://irsa-fornax-testdata/ZTF/dr23/objects/hats'),
+        object_margin_cache=None,
+        lightcurve_catalog_path=UPath('s3://irsa-fornax-testdata/ZTF/dr23/lc/hats'),
+        lightcurve_margin_cache=None,
+        object_columns=["oid", "ra", "dec"],
+        light_curve_columns=["objectid", "hmjd", "filterid", "mag", "magerr", "catflags"],
+        id_col='objectid',
+        time_col='hmjd',
+        flux_col='mag',    # this becomes raw.data['flux']
+        err_col='magerr',  # this becomes raw.data['err']
+        radius=radius,
+        filter_id_to_name=None,
+    )
+    # 2) turn the MultiIndex into columns
+    df = raw.data.reset_index()
+
+    # 3) rename columns
+    df = df.rename(columns={"hmjd": "time"})
+    #these are called flux and err in the generic hats_get_lightcurves, but are really magnitudes for ztf
+    df = df.rename(columns={'flux': 'mag', 'err': 'magerr'})
+
+    # 4) explode each array column into individual rows
+    df = df.explode(["time", "mag", "magerr"], ignore_index=True)
+    df = df.astype({
+        "time":   "float",
+        "mag":    "float",
+        "magerr": "float",
+    })
+
+    # 5) drop any epochs flagged as bad
+    # df = df.loc[df["catflags"] < 32768]
+
+    # 6) convert mag/magerr → flux/err (mJy)
+    mag    = df["mag"].to_numpy()
+    magerr = df["magerr"].to_numpy()
+    flux_up  = ((mag - magerr) * u.ABmag).to_value('mJy')
+    flux_low = ((mag + magerr) * u.ABmag).to_value('mJy')
+    df["flux"] = (mag * u.ABmag).to_value('mJy')
+    df["err"]  = (flux_up - flux_low) / 2
+
+    # 7) clean up intermediate columns (optional)
+    df = df.drop(columns=["mag", "magerr"])
+
+    # 8) re-index and return
+    df_lc = df.set_index(['objectid', 'label', 'band', 'time'])
+    
+    return MultiIndexDFObject(data=df_lc)
```

Comment:
```suggestion
    df = df.set_index(['objectid', 'label', 'band', 'time'])
    
    return MultiIndexDFObject(data=df)
```

Use the same name to avoid making a copy.